Updated Multilingual added Sub Docs

marcorensch opened the pull request:
### **User description**

1. Created Language Files page based on the original https://docs.joomla.org/Creating_a_language_definition_file including updates and examples.
2. Created Plural page where plural language translations got explained including examples. 
3. Created Page to explain the usage of variables in language translations and definition of custom placement for language variables including examples.
4. Created Format Specifiers page based on the original https://docs.joomla.org/Formatted_fields_in_language_translation_strings including updates and examples.
5. Moved everything as Sub Pages of the Multilingual page.


___

### **PR Type**
Documentation


___

### **Description**
- Created new documentation page explaining Joomla language files implementation and best practices
- Detailed explanation of language key naming conventions and structure (COM/MOD/PLG prefixes)
- Covered value specifications including proper quoting and HTML usage guidelines
- Added section about common errors in language files with examples
- Included information about file locations and updating process during development



___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>index.md</strong><dd><code>New documentation for Joomla language files implementation</code></dd></summary>
<hr>

docs/building-extensions/language-files/index.md

<li>Created comprehensive guide for Joomla language files<br> <li> Explained language key and value conventions and specifications<br> <li> Provided examples of common errors and best practices<br> <li> Added information about file locations and updating process<br>


</details>


  </td>
  <td><a href="https://github.com/joomla/Manual/pull/355/files#diff-abebebd1a041b3845cb697131375e0b68edb738d3c0a5d6914a1d5dce4b8bdf2">+130/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

HLeithner commented on docs/building-extensions/language-files/index.md:
````diff
@@ -0,0 +1,130 @@
+the Language Keys like this:
+```ini
+COM_EXAMPLE_FONT_COLOR_FIELD_LABEL="Font-Color"
+COM_EXAMPLE_FONT_COLOR_FIELD_DESC="Select the Font-Color for xy"
````
Thats outdated, schould be DESCRIPTION not longer DESC

HLeithner commented on docs/building-extensions/language-files/index.md:
````diff
@@ -0,0 +1,130 @@
+```
+
+## Conventions & Specifications for Language Keys
+The `KEY` should only use **ASCII characters**. It should only contain capital letters, digits, underscores and hyphens,
````
I think core doesn't use hypens nor dots. I think shouldn't be a suggestion but also don't hurt.

HLeithner commented on docs/building-extensions/language-files/index.md:
```diff
@@ -0,0 +1,130 @@
+:::
+
+## Case Sensitivity when using with Text::
+When you use the key in a Text:: call, the case does not matter as strings are folded to upper case before searching
```
Would formulate it the way as above so that uppercase only is convention.

HLeithner commented on docs/building-extensions/language-files/index.md:
```diff
@@ -0,0 +1,130 @@
+If there is an error in a language file, the file is not parsed correctly and
+individual translations are missing. It is then helpful to examine the corresponding language file with a code editor
+with syntax highlighting that can process the INI file.
+### Unescaped Double-Quotes in the Value String
```
Before and haver headline a blank line should be insert

HLeithner commented on docs/building-extensions/language-files/index.md:
```diff
@@ -0,0 +1,130 @@
+searches there for the translations for the respective language. If changes are made to language files, the extension
+must be installed again or the language file must be placed manually in the corresponding folder.
+
+| What     | Where |
```
Markdown should also be readable and nice formated

HLeithner reviewed (commented):
Thanks looks like a good starting point.

I miss some bits, from my head I miss plurals and convents for cross component keys like jhelp and jtoolbar and some others.

marcorensch commented:
New Bunch of Changes - Removed first created page and moved everything under "Multilingual" - Added:
- Language Files Documentation
- Plural
- Variables
- sprintf Format Specifiers including tested examples (J! 5.2 - PHP 8.3)

HLeithner commented:
Nice, I'm not sure if Multilingual is the best term but ok for now I think.

Some notices from my site, Components shouldn't use the /languages or /administrator/languages folder, the component should be self contained, that makes it simpler to maintain, for example you don't need an extra section in the xml for this, instead you just add the `languages` folder to the copy reference like any other folder.

the plural part is more complex then just _1 _2, I don't know from my head but I think it could be per language different.

for english you have
_0
_ONE or _1
_OTHER or _MORE

it's also possible that you have _12 or _DOZEN depending on the language.

antoher thing is that we have more then Text::_ function we also have a Text::alt function for example... never the less you should never use Text::_ if possible because it's not context aware instead you should have a Language Object in place which allows you to use the correct language.

About the "Format Specifiers" I would initially notice that this function is basically sprintf from PHP and reference to the php manual and say that everything from the manual is also true for this function.

I didn't read it completely but it's important to note that sprintf with the wrong variable count would end in a fatal error which is (intentional) not catched by joomla.

marcorensch commented:
@HLeithner : Implemented suggested changes 

About: _"it's also possible that you have _12 or _DOZEN depending on the language."_
I've done a ton of researches and was quite sure about this too but it did not work anymore as far as i can see, i've tested "_DOZEN" / "_A_DOZEN" and "_ADOZEN" unsuccesfully...

I've moved the "Using Text::_" part out of the Multilingual page and created a subpage containing all the available options in an overview with a simple example.

HLeithner commented:
It's something that could be possible per language, for example a language could return _DOZEN for the value 12 but no language has it implemented (at least I'm not aware of).

for example Welsh has _MANY for 6 and _FEW for 3.

HLeithner commented:
I would like to thank you, you did a really great job, I merge this for now. it can be improved later.
